source/core_sntp_client.c

```diff
@@ -102,3 +121,293 @@ SntpStatus_t Sntp_Init( SntpContext_t * pContext,
 
     return status;
 }
+
+/**
+ * @brief Utility to calculate the difference in milliseconds between 2
+ * SNTP timestamps.
+ *
+ * @param[in] pCurrentTime The more recent timestamp.
+ * @param[in] pOlderTime The older timestamp.
+ *
+ * @note This functions supports the edge case of SNTP timestamp overflow
+ * when @p pCurrentTime represents time in NTP era 1 (i.e. time since 7 Feb 2036)
+ * and the @p OlderTime represents time in NTP era 0 (i.e. time since 1st Jan 1900).
+ *
+ * @return Returns the calculated time duration between the two timestamps.
+ */
+static uint32_t calculateElapsedTimeMs( const SntpTimestamp_t * pCurrentTime,
+                                        const SntpTimestamp_t * pOlderTime )
+{
+    uint32_t timeDiffMs = 0U;
+
+    assert( pCurrentTime != NULL );
+    assert( pOlderTime != NULL );
+
+    timeDiffMs = ( pCurrentTime->seconds - pOlderTime->seconds ) * 1000U;
+
+    if( pCurrentTime->fractions > pOlderTime->fractions )
+    {
+        timeDiffMs += ( pCurrentTime->fractions - pOlderTime->fractions ) /
+                      ( SNTP_FRACTION_VALUE_PER_MICROSECOND * 1000U );
+    }
+    else
+    {
+        timeDiffMs -= ( pOlderTime->fractions - pCurrentTime->fractions ) /
+                      ( SNTP_FRACTION_VALUE_PER_MICROSECOND * 1000U );
+    }
+
+    return timeDiffMs;
+}
+
+/**
+ * @brief Sends SNTP request packet to the passed server over the network
+ * using transport interface's send function.
+ *
+ * @note For the cases of partial or zero byte data transmissions over the
+ * network, this function repeatedly retries send operation by calling the
+ * transport interface until either:
+ * 1. The requested number of bytes @p packetSize have been sent.
+ *                    OR
+ * 2. No byte cannot be sent over the network for the
+ * #SNTP_SEND_RETRY_TIMEOUT_MS duration.
+ *                    OR
+ * 3. There is an error in sending data over the network.
+ *
+ * @param[in] pNetworkIntf The UDP transport interface to use for
+ * sending data over the network.
+ * @param[in] timeServer The IP address of the server to send the
```

Comment:
Nit: Since in other documentation blocks it's specified as v4
```suggestion
 * @param[in] timeServer The IPv4 address of the server to send the
```